manual/tools/parse_doxygen_xml.py

```diff
@@ -503,9 +503,15 @@ def generate_params(self, param_string):
             """ % param_string
         return ""
 
+    def make_nbsp(self, string):
+        """Make all spaces non-breaking"""
+
+        # Use unicode non-breaking space. HTML &nbsp; is not valid in all Markdown flavours.
+        return string.replace(" ", "\u00a0")
+
     def generate_param_string(self, param_info, param_name):
         return "`%(type)s` | `%(name)s` | %(desc)s\n" % {
-            "type": self.get_text(param_info["type"], escape=False),
+            "type": self.make_nbsp(self.get_text(param_info["type"], escape=False)),
```

Comment:
Not sure I like this, what happens if you copy and paste these lines? I don't see the problem of breaking lines.

Reply:
Without it, the line breaks like this even on wide screens:

```
seL4_Word | next column
*         |
```

Reply:
Let me check what happens when I copy the rendered text

Reply:
It gets copied as normal space from the browser, so should be fine for copy/pasting type info out of the docs.

Reply:
> Without it, the line breaks like this even on wide screens:

Well, that's a sign our text is too wide. Perhaps we have too much indentation?

Reply:
It's just markdown table layout -- the width of columns can't be controlled from just markdown, it depends on the size of the other columns and descriptions are long, so they will always overpower short type columns.

We could try to insert explicit layout, but then we should really be generating html instead of markdown.

Reply:
I looked on the API website and see what you mean. Non-breaking spaces is probably as good as it gets.